ecscollector/collector.go

```diff
@@ -149,10 +165,16 @@ type collector struct {
 }
 
 func (c *collector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- taskMetadataDesc
+	ch <- taskCpuLimitDesc
+	ch <- taskMemLimitDesc
```

Comment:
The fact that these preexisting metrics worked without being described makes me wonder what the point of `Describe` is.

Reply:
Yes, `Describe()` is somewhat useless part of the API. It's not used externally by the metrics scrape. The use is mostly when using a Go package as a library and you want to expose information about metrics to a UI that is not a scrape.

One day I would like to actually make use of this as part of a more generalized exporter collector API and a UI in the exporter-toolkit. This would make it easier to take our various collector packages and build a "super exporter" / "agent" style multi-exporter.